### PR TITLE
Introduce: si seq prune

### PR DESCRIPTION
**Pull request by MichalCz**
There's not much to it. Upload a couple sequences and then:

```sh
si seq ls
si seq prune
si seq ls
```

The second `ls` should show... no sequences? :)

### CODE REVIEW

**Review by ErykSol — CHANGES_REQUESTED** [review]

File: packages/cli/src/lib/commands/sequence.ts [hunk 1]
```diff
@@ -69,6 +69,30 @@ export const sequence: CommandDefinition = (program) => {
         .description("create archived file (package) with sequence for later use")
         .action((path, { stdout, output }) => packAction(path, { stdout, output }));
 
+    sequenceCmd
```
Comment (ErykSol) [inline packages/cli/src/lib/commands/sequence.ts:72]: At the end of this file is already prepared command for prune in developer mode- this should be moved down and merged with that one (remove developer flag)  to keep proper command order.

**Review by ErykSol — CHANGES_REQUESTED** [review]

**Review by ErykSol — APPROVED** [review]

File: packages/cli/src/lib/commands/sequence.ts [hunk 2]
```diff
@@ -69,6 +69,40 @@ export const sequence: CommandDefinition = (program) => {
...
+        .command("prune")
```
Comment (a-tylenda) [inline packages/cli/src/lib/commands/sequence.ts]:
```suggestion
        .command("prune")
        .description("delete multiple sequences on actual selected hub")
```

**Comment by a-tylenda** [comment]
Verification failed during command execution: `si seq prune --force`.
**Steps to Reproduce**
1. start sth: `scramjet-transform-hub`
2. send a few Sequences to STH, start the last one: `si seq start -`
3. force prune to delete all Sequences including the one which is running: `si seq prune --force`

Screenshot of error I received:

image

I think that there is not enough time for the Sequence to stop working after "kill" is sent. DELETE request is sent while killing process is not finished.

File: packages/cli/src/lib/commands/sequence.ts [hunk 3]
```diff
@@ -140,15 +140,37 @@ export const sequence: CommandDefinition = (program) => {
...
+                        displayMessage(`Sequence ${seq.id} has running instances, use --force to kill those.`);
+                        continue;
+                    }
+                    // maybe we should stop
+                    displayMessage(`Killing instances of sequence ${seq.id}`);
+                    await Promise.all(
+                        seq.instances.map(async (id) => {
+                            if (lastInstanceId === id) sessionConfig.setLastInstanceId("");
+
+                            return getInstance(id).kill();
```
Comment (a-tylenda) [inline packages/cli/src/lib/commands/sequence.ts:166]:
```suggestion
                            return getInstance(id).kill();
                            defer(1000);
```

maybe we could give some time for Sequence to stop working?

File: packages/cli/src/lib/commands/sequence.ts [hunk 4]
```diff
@@ -140,15 +173,43 @@ export const sequence: CommandDefinition = (program) => {
...
-            .description("delete multiple sequences on actual selected hub")
-            .action(() => {
-            // FIXME: implement me
-                throw new Error("Implement me");
-            });
+    sequenceCmd
+        .command("prune")
+        // .option("--all")
+        // .option("--filter")
+        .option("-f,--force", "Removes also active sequences")
```
Comment (a-tylenda) [inline packages/cli/src/lib/commands/sequence.ts:180]:
```suggestion
        .option("-f,--force", "Removes also active sequences")
        .description("remove all Sequences from the current scope (use with caution)")
        
```

**Comment by a-tylenda** [comment]
verification complete:

image

**Pull request merged**